src/components/molecules/TableHeader/components/EditTableTitleModal/EditTableTitleModal.tsx

```diff
@@ -1,55 +1,125 @@
+  onSave: (values: EditTableForm) => void;
+}
+
+export interface EditTableForm {
+  title: string;
+  description?: string;
+  capacity: number;
 }
 
 export const EditTableTitleModal: React.FC<EditTableTitleModalProps> = ({
   isShown,
   title,
+  description,
+  capacity,
   error,
   onSave,
   onHide,
   onCancel,
 }) => {
-  const [newTableTitle, setNewTableTitle] = useState<string>(title);
+  const {
+    register,
+    handleSubmit,
+    errors,
+    // formState,
+    // setError,
+    // clearError,
+    watch,
+  } = useForm<EditTableForm>({
+    mode: "onChange",
+    reValidateMode: "onChange",
+    defaultValues: {
+      title,
+      description,
+      capacity,
+    },
+  });
 
-  const saveChanges = useCallback(() => {
-    onSave(newTableTitle);
-    onHide();
-  }, [newTableTitle, onHide, onSave]);
+  const values = watch();
 
-  const changeTableTitle = useCallback(
-    (e: React.ChangeEvent<HTMLInputElement>) => {
-      setNewTableTitle(e.target.value);
-    },
-    []
-  );
+  const onSubmit = useCallback(() => {
+    onSave(values);
+    onHide();
+  }, [onHide, onSave, values]);
 
   return (
     <Modal show={isShown} onHide={onHide}>
       <Modal.Body>
-        <input value={newTableTitle} onChange={changeTableTitle}></input>
-        {error && <label className="input-error">{error}</label>}
-        <div className="edit-title-buttons">
-          <button className="btn btn-centered btn-secondary" onClick={onCancel}>
-            Cancel
-          </button>
-          <button
-            className="btn btn-centered btn-primary"
-            onClick={saveChanges}
-          >
-            Save Changes
-          </button>
-        </div>
+        <form onSubmit={handleSubmit(onSubmit)} className="edit-table-modal">
+          <InputField
+            ref={register({ required: true })}
+            name="title"
+            containerClassName="edit-table-modal__input--spacing"
+            placeholder="Table topic"
+          />
+          {errors.title && errors.title.type === "required" && (
+            <span className="input-error">Title is required</span>
+          )}
+          <InputField
+            ref={register}
+            name="description"
+            containerClassName="edit-table-modal__input--spacing"
+            placeholder="Describe this table (optional)"
+          />
+          <div className="edit-table-modal__capacity">
+            <label className="edit-table-modal__max-capacity">
+              Number of seats (max {MAX_TABLE_CAPACITY})
+            </label>
+            <InputField
+              ref={register({
+                required: true,
+                max: MAX_TABLE_CAPACITY,
+                min: MIN_TABLE_CAPACITY,
+              })}
+              name="capacity"
+              type="number"
+              min={MIN_TABLE_CAPACITY}
+              max={MAX_TABLE_CAPACITY}
+              placeholder="Max seats"
+            />
+          </div>
```

Comment:
The `InputField` should be nested inside the `label`:

- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

Otherwise we need to use `id`s to associate them.

Reply:
Do you need to set min/max in both `register` and directly on the props? Will `register` not handle both for us?

Reply:
Yes, there is a carousel type of thing when `type='number'` And this adds validation for that too.

Reply:
Fair enough, makes sense. I just assumed it would set those aspects automagically for both.

Reply:
Strangely it doesn't. I suppose because the register validates only on submit, where the carousel is input's html element and that's why it requires another validation.

Reply:
Looking at: 

- https://react-hook-form.com/get-started#Applyvalidation

> React Hook Form makes form validation easy by aligning with the existing HTML standard for form validation.

- https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation

What they mean by 'aligning' here is ambiguous though. Would need to test the lib deeper to see exactly how it operates/etc.

If you comment out the extra html props you added, and then inspect the runtime element, does it have those props set in the DOM automagically from `register`? And if not, if you change the validate mode back to one of the 'live' variations, will it do so then?

---

> I suppose because the register validates only on submit, where the carousel is input's html element and that's why it requires another validation.

If you changed it to validate on change/similar, and comment out the extra html props you added, would it work as expected?

Reply:
Searching through their backlog of issues, looks like it might have supported native validation in v3, but has been removed from v4:

- https://github.com/react-hook-form/react-hook-form/issues/733#issuecomment-569413490
- https://github.com/react-hook-form/react-hook-form/pull/666
  - > API to be removed list:
    - > native validation: hardly anyone used this feature. https://react-hook-form.com/api/#Browserbuiltinvalidation
    
Changelog confirms:

- https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md#removed-1

---

**tl;dr**: Seems like you're definitely correct in that it doesn't automatically add the native browser constraints anymore.

A different/better question here now might be whether we want to have both the native and the custom props configured, rather than just using one or the other?

Reply:
We should check if something like that exists otherwise this will be difficult to deal with. I'll see if I can find something.

Reply:
Looks like we should try and use only the `useForm` validations and just disable the native UI element. That way we can avoid the secondary validation.

Reply:
In this case the `type="number"` prop of the input renders two arrows for increment and decrement. If we disable this with CSS then we won't have to natively validate it.

Reply:
This is a rare case because of this numeric input, that's why in other cases this was never an issue.

Reply:
Instead of disabling it, can we just set our hook based validation to run when it changes or similar? Disabling convenience features like that seems like something we probably shouldn’t be doing?